File: doc_mat/src/main/java/edu/esprit/entities/Matiere.java
package edu.esprit.entities;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Objects;

public class Matiere {
     private int id;
   private String nommatiere;
   private String description;
   private  String annee ;
   private CAT categorie;

    public Matiere(){ }
     public  Matiere(int id){
         this.id=id;
     }

    public Matiere(String nommatiere, String description , String annee , CAT categorie) {
        this.nommatiere = nommatiere;
        this.description = description;
        this.annee=annee;
        this.categorie=categorie;
    }

    public Matiere(int id, String nommatiere, String description, String annee , CAT categorie) {
        this.id = id;
        this.nommatiere = nommatiere;
        this.description = description;
        this.annee=annee;
        this.categorie=categorie;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNommatiere() {
        return nommatiere;
    }

    public void setNommatiere(String nommatiere) {
        this.nommatiere = nommatiere;
    }

    public String getDescription() {
        return description;
    }

    public String getAnnee() {
        return annee;
    }

    public void setAnnee(String annee) {this.annee = annee;}

    public CAT getCategorie() {
        return categorie;
    }

    public void setCategorie(CAT categorie) {
        this.categorie = categorie;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Matiere{" +
                "id=" + id +
                ", nommatiere='" + nommatiere + '\'' +
                ", description='" + description + '\'' +
                 '\'' +" annee :" + annee+
                '\'' +" categorie :" + categorie +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Matiere matiere = (Matiere) o;
        return Objects.equals(nommatiere, matiere.nommatiere);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nommatiere);
    }
}

File: projettest-main2/src/Entity/Matiere.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\MatiereRepository;

#[ORM\Entity(repositoryClass: MatiereRepository::class)]

class Matiere
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id=null;

    #[ORM\Column(length: 255)]
    private  ?string $nomMatiere;
    #[ORM\Column(length: 255)]
    private  ?string $description;

    #[ORM\Column(length: 255)]
    private ?string $annee;

    #[ORM\Column(length: 255)]
    private ?string  $categorie;

    #[ORM\ManyToOne(targetEntity:Utilisateurs::class , inversedBy:' matieres')]
    private ?Utilisateurs $idu;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomMatiere(): ?string
    {
        return $this->nomMatiere;
    }

    public function setNomMatiere(string $nomMatiere): static
    {
        $this->nomMatiere = $nomMatiere;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getAnnee(): ?string
    {
        return $this->annee;
    }

    public function setAnnee(string $annee): static
    {
        $this->annee = $annee;

        return $this;
    }

    public function getCategorie(): ?string
    {
        return $this->categorie;
    }

    public function setCategorie(string $categorie): static
    {
        $this->categorie = $categorie;

        return $this;
    }

    public function getIdu(): ?Utilisateurs
    {
        return $this->idu;
    }

    public function setIdu(?Utilisateurs $idu): static
    {
        $this->idu = $idu;

        return $this;
    }
    public function __toString()
    {
        return $this->getId();
    }


}